### ISSUE TITLE
NixOs tests are currently failing on develop

### ISSUE THREAD

**Issue by peeech** [Holo-Host/holo-nixpkgs#1081]
Both of our NixOs tests located in [/tests](https://github.com/Holo-Host/holo-nixpkgs/tree/develop/tests) folder are failing on develop branch.

Current tip of develop is at https://github.com/Holo-Host/holo-nixpkgs/commit/71d62aa309ae5379a72eb4198e61cf676005aad3

**Comment by peeech** [comment]
After running tests locally in a interactive mode I can see that tests are failing because VM is never able to boot up past `testScript.start_all()`.

I am seeing in the logs:
```
machine # [  274.411895] lair-keystore-pre-start[1220]: Lair Keystore Prestarting...
machine # [  274.496144] lair-keystore-pre-start[1220]: initializing keystore from config at /etc/hpos/config.json with password pass
machine # [  284.654008] lair-keystore-pre-start[1229]: Error: failed to open file /etc/hpos/config.json
machine # [  284.680196] lair-keystore-pre-start[1229]: Caused by:
machine # [  284.697170] lair-keystore-pre-start[1229]:     No such file or directory (os error 2)
```

**Comment by peeech** [comment]
This seems to be caused by the fact that `/tests/hpos-api-tests` is requiring first starting VM in [line 17](hpos-api-tests), then creates `/etc/hpos/config.json` file in [line 22](https://github.com/Holo-Host/holo-nixpkgs/blob/develop/tests/hpos-api-tests/default.nix#L22). But VM at boot time is trying to start [`lair-keystore.service`](https://github.com/Holo-Host/holo-nixpkgs/blob/develop/profiles/logical/hpos/sandbox/default.nix#L96), which in turn is trying to access not-yet-created `/etc/hpos/config.json` and fails, VM is never able to start.

**Comment by alastairong** [comment]
Strange that these tests passed on their respective PRs...

**Comment by peeech** [comment]
It does not seem possible to create a file before VM is up and running. One solution I see can be to create a service which runs before `lair-keystore.service` that creates `/etc/hpos/config.json` in the same manner as [line 22](https://github.com/Holo-Host/holo-nixpkgs/blob/develop/tests/hpos-api-tests/default.nix#L22). This service would be only active on a [sandbox profile](https://github.com/Holo-Host/holo-nixpkgs/blob/develop/profiles/logical/hpos/sandbox/default.nix).

**Comment by zo-el** [comment]
@peeech I am not able to see where the tests were failing on hydra. it seems like it's failing on the `hydra-dev` but not on `hydra`.

Also on boot lair-keystore will fail to start, but the lair service just keeps retrying and when it finds the file it is able to spin up.

This is the log of one of the most recently develop: https://cache.holo.host/log/hn9ksjg3sayrx54qb8442cw6hzlr1mgh-vm-test-run-systemd-resilience-test.drv
You can see the test failing to start lair first and then spins it up later and therefore it suceeds.

**Comment by peeech** [comment]
Let's look at [latest builds](https://hydra.holo.host/jobset/holo-nixpkgs/develop) of develop, namely [1838529](https://hydra.holo.host/eval/1838529) where both hpos tests are failing. If you look at the [log of build 51427](https://cache.holo.host/log/qk2sxpv6iab057847fddyr154nbgd9rj-vm-test-run-systemd-resilience-test.drv) you will see that what happens is:
- lair will fail to pre-start because there's no config file (at 46.014214)
- because of that envoy can't start (at  49.362229)
- because of that holochain can't start (at 50.157290)
- because of that configure-holochain can't start (at 50.239195)
- because of that holo-auto-installer can't start (at 50.311360)
- then eventually lair will start (at 66.617368)
- but it's too late because entire dependency pyramid is already broken
- and the rest of services fails to start and job fails

This issue is not present on a real holoport, because holo-init activates those services only AFTER config file is created. And this is exactly what [mock-hpos-config](https://github.com/Holo-Host/holo-nixpkgs/pull/1083/commits/82b8dcd641809091a31300eecf6bddcea74cbb86) is doing on sandbox profile.

You can see it in action on #1075 where `mock-hpos-config` fix was already implemented. For example on [latest build](https://hydra.holo.host/build/52254) test is passing with no errors at all in 262 seconds.

**Comment by zo-el** [comment]
Switching from waiting for lair-keystore to waiting for holochain should solve that issue. Check this out: #1091

**Comment by peeech** [comment]
fixed in #1161
